plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+        """
+
+        try:
+            payload = {"payload": collected_add_vn_payload}
+            task_name = "add_layer3_virtual_networks"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
+                self.set_operation_result("failed", False, self.msg, "ERROR")
+                return self
+
+            success_msg = "Layer3 Virtual Network(s) '{0}' created successfully in the Cisco Catalyst Center.".format(self.created_vn)
+            self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg)
+
+        except Exception as e:
+            self.msg = (
+                "An exception occured while creating the layer3 Virtual Network(s) '{0}' in the Cisco Catalyst "
+                "Center: {1}"
+            ).format(self.created_vn, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR")
+
+        return self
+
+    def is_vn_needs_update(self, vn_details, vn_in_ccc):
+        """
+        Determines if a Virtual Network requires an update based on its details and current state in
+        the Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            vn_details (dict): A dictionary containing details about the virtual network, including:
+                - vn_name (str): The name of the virtual network.
+                - fabric_site_locations (list): A list of fabric site locations associated with the virtual network.
+                - anchored_site_name (str): The name of the anchored site.
+            vn_in_ccc (dict): A dictionary representing the current state of the virtual network in the Cisco
+                            Catalyst Center, which includes:
+                - fabricIds (list): A list of fabric IDs currently associated with the virtual network.
+                - anchoredSiteId (str): The ID of the anchored site currently associated with the virtual network.
+        Returns:
+            bool: Returns `True` if the virtual network needs an update; otherwise, returns `False`.
+        Description:
+            This function checks if a virtual network needs to be updated by comparing its provided details
+            with the existing configuration in the Cisco Catalyst Center.
+            If all checks are passed without indicating an update, the function returns `False`.
+        """
+
+        vn_name = vn_details.get("vn_name")
+        fabric_ids_in_ccc = vn_in_ccc.get("fabricIds")
+        fabric_locations = vn_details.get("fabric_site_locations")
+
+        if fabric_locations is None:
+            self.log("There is no fabric site details given in the playbook for the vn '{0}'.".format(vn_name), "INFO")
+            return False
+
+        if not fabric_locations and fabric_ids_in_ccc:
```

Comment:
self.log("Fabric locations provided, but no fabric IDs found for VN '{0}'.".format(vn_name), "INFO")